### ISSUE TITLE
`Roman.Numeral.transpose()` partially doesn't work.

### ISSUE THREAD

**Issue by malcolmsailor** [cuthbertLab/music21#1413]
**music21 version**

`8.0.0a12`   

**Problem summary**

```
In [29]: RN = music21.roman.RomanNumeral

In [30]: RN("I", "C").transpose(4).pitches # Looks good!
Out[30]: (<music21.pitch.Pitch E4>, <music21.pitch.Pitch G#4>, <music21.pitch.Pitch B4>)

In [31]: RN("I", "C").transpose(4) # oops, key should be E major
Out[31]: <music21.roman.RomanNumeral I in C major>

In [32]: RN("I", "C").transpose(4).key # oops again
Out[32]: <music21.key.Key of C major>
```

Note that if the roman numeral is embedded in a part, the keys in the part are transposed but not the keys of the roman numeral:

```
In [34]: rntxt = """Time Signature: 4/4
    ...: m1 C: I"""

In [37]: converter.parse(rntxt, format="romanText").transpose(4).show("text")
{0.0} <music21.metadata.Metadata object at 0x1076f61f0>
{0.0} <music21.stream.Part 0x10429e400>
    {0.0} <music21.stream.Measure 1 offset=0.0>
        {0.0} <music21.key.Key of E major>
        {0.0} <music21.meter.TimeSignature 4/4>
        {0.0} <music21.roman.RomanNumeral I in C major>
```

The romanText writer seems to rely on the RomanNumeral keys, so it then ignores the transposition. I.e., after running `converter.parse(rntxt, format="romanText").transpose(4).write("romanText", "rntxttemp")` the contents of "rntxttemp" are:

```
Composer: Composer unknown
Title: Title unknown
Analyst: 
Proofreader: 

Time Signature: 4/4
m1 C: I
```

Note that the key is C and not E!

I can have a stab at fixing this. It seems to me that all that should be necessary is to call `transpose()` on the keys within the RomanNumerals.

Incidentally, in making this issue I noticed that `.show("romanText")` doesn't seem to work (I would have expected it to just print out the romanText representation, analogous to `.show("text")`).

**Comment by mscuthbert** [comment]
If you transpose I in C major up a P5, does it become I in G major or V in C major?

It all depends on the context what the correct answer should be.  If you're transposing the whole piece, then just the key should be transposed.  But then there's another problem: I frequently share Key objects between roman numerals:

```
e = key.Key('E')
i1 = roman.RomanNumeral('I', e)
i2 = roman.RomanNumeral('I', e)

for rn in [i1, i2]:
    rn.transpose('P5')
    print(rn)
```

Now we'd get something like:  I in B, I in F# -- AND at the end, they'd both be I in F#.  

This is why transpose has not been implemented for RomanNumerals yet.

**Comment by malcolmsailor** [comment]
OK, I understand your reservations, although for my part I think the ordinary answer to

> If you transpose I in C major up a P5, does it become I in G major or V in C major?

is I in G major, and that the latter would better be implemented by a function called "diatonic_transpose" or "transpose_within_key" or something like that.

However I would suggest that if you don't want to implement transpose for RomanNumerals then that method should raise an Exception or at the very least a warning. Regardless of whether C major transposed by a 5th should be I in G major or V in C major, I think we can be confident that it should *not* be I in C major.

**Comment by mscuthbert** [comment]
I want to do it, I just want to take the time to consider the various possibilities.

Like if we call transpose() on a stream, we probably want to change the keys of the RNs, but we need a mechanism to make sure that it doesn't happen more than once for the same key object.  For instance we could have a `_keyMemo` private set of ids of keys already seen that and skip any keys visited more than once, etc.

**Comment by malcolmsailor** [comment]
Right, I was thinking about a memo too. But it seems that even that couldn't avoid the issue you mention in your previous post: if the user iteratively transposes individual roman numerals that share a key, the key will end up being transposed multiple times. The only solution I can see would be to have roman numeral transposition always create a new key object. I can imagine this might be undesirable for other reasons however. Perhaps a warning would be appropriate the first time the user calls RomanNumeral.transpose()?

**Comment by mscuthbert** [comment]
>  The only solution I can see would be to have roman numeral transposition always create a new key object.

That's probably a great idea.

**Comment by malcolmsailor** [comment]
OK, so shall I update my PR to have roman numeral transposition create a new key object each time? I will write a few new tests for the cases you have brought up as well.

**Comment by malcolmsailor** [comment]
Closed by #1414
